src/modules/image-upload/image-upload.js

```
@@ -23,39 +23,74 @@ function getExpiration() {
 	return (new Date(Date.now() + validity)).toISOString();
 }
 
+// For old bucket
+function getOldExpiration() {
+	const validity = config.oldS3.validity;
+	return (new Date(Date.now() + validity)).toISOString();
+}
+
 function sign(key, data) {
 	return crypto.createHmac('sha256', key).update(data).digest();
 }
 
 function getKeyPrefix(userId, uploadType, textId) {
 	switch (uploadType) {
 	case 'avatar':
+		return uploadType + '/' + userId + '/';
+	case 'content':
+		return uploadType + '/' + userId + '/' + textId + '/';
+	}
+	return '';
```

throw an error instead of returning empty string